test_app/tests/activitystream/test_api.py

```diff
@@ -33,3 +34,39 @@ def test_activitystream_api_read_only(admin_api_client, user):
     assert response.status_code == 405
     response = admin_api_client.delete(url)
     assert response.status_code == 405
+
+
+@pytest.mark.parametrize(
+    "has_rbac_app,who",
+    [
+        (True, "user"),
+        (False, "admin"),
+    ],
+)
+def test_activitystream_api_permission_classes(admin_api_client, user_api_client, has_rbac_app, who, settings):
+    """
+    Test that access to the activity stream is dynamically determined based on
+    whether or not RBAC is enabled.
+
+    If RBAC is enabled, then it locks down permissions on its own, so we allow IsAuthenticated.
+    If RBAC is not enabled, then we require IsSuperuser.
+    """
+    url = reverse("activitystream-list")
+
+    if 'ansible_base.rbac' in settings.INSTALLED_APPS:
+        if not has_rbac_app:
+            settings.INSTALLED_APPS.remove('ansible_base.rbac')
```

Comment:
Could you use `override_settings` to remove it... less destructively?

Reply:
This is using the pytest [settings](https://pytest-django.readthedocs.io/en/latest/helpers.html#settings) fixture, so it reverts automatically